### PR TITLE
Add product images block to product editor

### LINKED ISSUE #37272: Add images block to product block editor
**Description**

Merchants should have the ability to add images to a product. This issue will focus on adding the images block to the product block editor.
We should look into re-using the current images block within Gutenberg if possible, otherwise create a wrapper block component.

This issue will include the addition of a new Images section and description ( this should only require updating the product template ).

**Design**

Image

AkVNGImLgSqCObTQ3idVn7-fi-207-27099

**Acceptance criteria**

- [ ] A new images section should be available containing an image block
- [ ] Merchants/users should have the ability to add/remove images from a product and save.

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

* Adds the product images block to the new product block editor
* Allows html within section block descriptions
Screen Shot 2023-03-22 at 6 01 03 PM

Closes #37272 .

### How to test the changes in this Pull Request:

- [x] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Install the WooCommerce Beta Tester plugin. 
2. Go to `Tools > WCA Test Helper > Features`.
3. Enable the product blocks feature.
4. Navigate to the `Products > Add new`.
5. See that the images block exists on this page.
6. Try adding, replacing, and setting an image as the cover when multiple images exist.
7. Save or Add the product.
8. Make sure the product data is persisted on refresh.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/product-editor/src/components/section/edit.tsx [hunk 1]
```diff
@@ -2,8 +2,18 @@
...
 import { createElement } from '@wordpress/element';
-import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';
 import type { BlockEditProps } from '@wordpress/blocks';
+import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';
+import { sanitize } from 'dompurify';
+
+const ALLOWED_TAGS = [ 'a', 'b', 'em', 'i', 'strong', 'p', 'br' ];
+const ALLOWED_ATTR = [ 'target', 'href', 'rel', 'name', 'download' ];
+
+function sanitizeHTML( html: string ) {
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/section/edit.tsx]: I think we can move this helper function to its own folder so we can reuse it in other places.

**Review by mdperez86 — DISMISSED** [review]
LGTM!!! Nice job @joshuatf. I just left one comment that is not mandatory to do.

File: packages/js/product-editor/src/components/section/edit.tsx (hunk 1 above)
Comment (joshuatf) [reply packages/js/product-editor/src/components/section/edit.tsx]: Good idea!  Moved in https://github.com/woocommerce/woocommerce/pull/37455/commits/599b658d12dc3b8feafc45ef7accca148c94cd0d

**Comment by joshuatf** [comment]
@mdperez86 Moved the sanitize to a utils so we can reuse it.  Could you give this another ✅ ?

**Review by mdperez86 — APPROVED** [review]
🙌 Good job!

**Pull request merged**